pages/science/reactors/extensiblereactor.rst

```diff
@@ -0,0 +1,181 @@
+
+There are two "sides" to each of these equations: the terms left of the equals
+sign and the terms to the right of the equals sign. This is the format
+in which the user will be editing the governing equations. For example,
+if the user wishes to add a term for a large mass (say a rock) inside
+the reactor to see the effects on reaction temperature:
+
+.. math::
+
+   m c_p \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+  \tag{2}
+
+Will change to:
+
+.. math::
+
+   m c_p \frac{dT}{dt} + m_{rock} c_{p,rock} \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+
+A simple example is shown below to illustrate the process for implementing
+changes in Cantera's existing governing equations.
+We will be replacing the right-hand side (RHS) and left-hand side (LHS) of 
+the temperature governing equation for an Ideal Gas Constant Pressure Reactor.
+All other governing equations defining an Ideal Gas Constant Pressure Reactor
+will remain as the default.
+
+In this example
+
+.. math::
+
+   m c_p \frac{dT}{dt} = - \dot{Q} - \sum_k h_k \dot{m}_{k,gen}
+       + \sum_{in} \dot{m}_{in} \left(h_{in} - \sum_k h_k Y_{k,in} \right)
+  \tag{2}
+
+Will change to:
+
+.. math::
+
+   m_{rock} c_{p,rock}\frac{dT}{dt} + m_{gas}\frac{dT}{dt} = - \dot{Q}
+
+The governing equations will be modified through the user created Python class' methods.
+For each method, the name should be prefixed with **before_**, **after_**, or 
+**replace_**, indicating whether the this method should be called before, after, 
+or instead of the corresponding method from the base class.
+
+.. code-block:: python
+
+  #1 Define objects, properties, and initial conditions.
+
+   #create gas object
+   gas = ct.Solution('h2o2.yaml')
+   gas.TPX = 500, ct.one_atm, 'H2:2,O2:1,N2:4'
+
+   #define properties of gas and solid
+   mass_gas = 20 #[kg]
+   Q = 100 #[J/s]
+   mass_rock = 10 #[kg]
+   cp_rock = 1.0 #[J/kgK]
+
+   #initialize time at zero
+   time = 0 #[s]
+   n_steps = 300
+
+  #2 Define a new custom Reactor class. Here we named it "DummyReactor" and 
+   #chose the Ideal Gas Constant Pressure Reactor as the base class to inheret
+   #governing equations from. 
+
+   #define a class representing reactor with a solid mass and gas inside of it
+   class RockReactor(ct.DelegatedIdealGasConstPressureReactor):
+
+      #modify energy equation to include solid mass in reactor
+      
+      #after the initial solution for time t is computed ask Cantera to solve the modified 
+      #equation. The index 1 refers to modification of governing equation 2 in the reactor
+      #documentation (recall that indexing begins at 0).
+
+      def after_eval(self, t, LHS, RHS):
+      #although the time variable t is not used directly in the method definition it is a 
+      #required argument for the internal solver.
```

Comment:
The comment here should be indented like the body of the function, not flush with the `def`.